ipv8/lazy_community.py

```diff
@@ -35,7 +35,9 @@ def wrapper(self, source_address, data):
                 raise PacketDecodingError("Incoming packet %s has an invalid signature" %
                                           str([payload_class.__name__ for payload_class in payloads]))
             # PRODUCE
-            return func(self, Peer(auth.public_key_bin, source_address), *unpacked)
+            peer = self.network.verified_by_public_key_bin.get(auth.public_key_bin,
+                                                               Peer(auth.public_key_bin, source_address))
```

Comment:
The creation of a Peer object is not exactly cheap, and it happens on each lookup. If the speed is important here, it is better to create a Peer instance after the unsuccessful lookup only:
```python
peer = self.network.verified_by_public_key_bin.get(auth.public_key_bin) or \
       Peer(auth.public_key_bin, source_address))
```
or
```python
peer = self.network.verified_by_public_key_bin.get(auth.public_key_bin)
if peer is None:
    peer = Peer(auth.public_key_bin, source_address))
```

Reply:
Thanks 👍, good point: I'll change that.